core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala

```diff
@@ -65,7 +66,35 @@ trait WorkspaceApiService extends UserInfoDirectives {
           }
         }
     } ~
+      path("workspaces" / "tags") {
+        parameter('q.?) { queryString =>
+          get {
+            complete {
+              workspaceServiceConstructor(userInfo).getTags(queryString)
+            }
+          }
+        }
+      } ~
+      path("workspaces" / "id" / Segment ) { workspaceId =>
+        get {
+          parameterSeq { allParams =>
+            traceRequest { span =>
+              complete {
+                workspaceServiceConstructor(userInfo).getWorkspaceById(workspaceId,
+                  WorkspaceFieldSpecs.fromQueryParams(allParams, "fields"), span)
+              }
+            }
+          }
+        }
+      } ~
       path("workspaces" / Segment / Segment) { (workspaceNamespace, workspaceName) =>
+        /* we enforce a 6-character minimum for workspaceNamespace, as part of billing project creation.
+           the previous "mc", "tags", and "id" paths rely on this convention to avoid path-matching conflicts.
+           we might want to change the first Segment above to a regex a la """[^/.]{6,}""".r
+           but note that would be a behavior change: if a user entered fewer than 6 chars it would result in an
+           unmatched path rejection instead of the custom error handling inside WorkspaceService.
+        */
+
```

Comment:
additionally, we have some unit tests that use a namespace of "ns" - all those tests would need to be updated if we path-matched on a 6-char minimum here.